src/figuresClasses.ts

```diff
@@ -1,19 +1,124 @@
-export interface Figure {
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
+
+function isPositive(...args: number[]): boolean {
+  for (let i = 0; i < args.length; i += 1) {
+    if (args[i] < 0) {
+      return false;
+    }
+  }
 
+  return true;
 }
 
-export class Triangle {
+function triangleSideChecker(...args: number[]): boolean {
+  const sortedSides: number[] = args.sort((a: number, b:number) => (
+    b - a
+  ));
+
+  if (sortedSides[0] >= sortedSides[1] + sortedSides[2]) {
+    return false;
+  }
+
+  return true;
+}
+export interface Figure {
+  shape: Shape;
+  color: Color;
 
+  getArea(): number;
 }
 
-export class Circle {
+export class Triangle implements Figure {
+  public shape: Shape = 'triangle';
+
+  public color: Color;
+
+  public a: number;
+
+  public b: number;
+
+  public c: number;
+
+  constructor(
+    color: Color,
+    a: number,
+    b: number,
+    c: number,
```

Comment:
you can add public in constructor and delete it from shape

```suggestion
  shape: Shape = 'triangle';
  
  constructor(
    public color: Color,
    public a: number,
    public b: number,
    public c: number,
```